src/manager/network-manager.c

```diff
@@ -1930,28 +1930,53 @@ int manager_parse_proxy_config(GHashTable **c) {
 int manager_generate_network_config_from_yaml(const char *file) {
         _cleanup_(g_string_unrefp) GString *wifi_config = NULL;
         _cleanup_(network_unrefp) Network *n = NULL;
-        int r;
+        _cleanup_(netdev_link_unrefp) NetDevLink *l = NULL;
+	_auto_cleanup_ IfNameIndex *p = NULL;
+	char cmd[256];
+	int r;
 
         assert(file);
+        snprintf(cmd, 256, "grep -qw link: %s", file);
 
-        r = parse_yaml_network_file(file, &n);
-        if (r < 0) {
-                log_warning("Failed to parse configuration file '%s': %s", file, strerror(-r));
-                return r;
-        }
+        if (string_has_suffix(file, "link.yaml") && !system(cmd)) {
```

Comment:
No it's wrong ``` 
  link: <============================
     name: eth1
     dhcp: ipv4
     dhcp-client-identifier: mac```

It it starts with link the link code should execute not a suffix. Suffix can be anything. Please rework